### ISSUE TITLE
False nick detection when it's a part of a word with `

### ISSUE THREAD

**Issue by christer88** [thelounge/lounge#2008]
![thelounge](https://user-images.githubusercontent.com/11909212/35480786-5df28c92-0416-11e8-92ed-0c067e22226c.png)

User S was linked and coloured when the word **_it's_** was written in channel. User S shouldn't have been detected since it was a part of a bigger word.

**Comment by christer88** [comment]
It's a tricky edge case since we would want user **_test_** to be detected in _**test's**_, but at the same time not user **_S_**. A fix for this would be to not detect users after a **_'_**?

**Comment by MaxLeiter** [comment]
@christer but what if I ping you as ‘christer’?

**Comment by realies** [comment]
Wrong tag, but what if there's a check which validates `'s` as the end of a word and skips highlighting?

**Comment by MaxLeiter** [comment]
No, meant it at him

**Comment by christer** [comment]
Me? don't think so.

**Comment by Jay2k1** [comment]
Apparently this issue is not limited to apostrophes, umlauts also do it.

![image](https://user-images.githubusercontent.com/2070503/36103047-8177892a-100e-11e8-92ee-b89631c9ccbf.png)

**Comment by ivan** [comment]
It does not make any sense to detect nicks in the middle of messages: I'm in a channel with a bot named `the`. Every `the` is highlighted. This is nonsense. The best fix here is to get rid of the feature.

**Comment by xPaw** [comment]
@ivan That's for your constructive feedback.

**Comment by Daniel15** [comment]
It would be good to have a user-editable blacklist for this, so users could add common words to it to avoid highlighting them. `##javascript` on FreeNode has the same issue - There's a user called `t`, so every mention of `t` after an apostrophe (eg. `can't`, `won't`), highlights the `t`.

**Comment by Daniel15** [comment]
What I mean is that some people may like the short nicks being highlighted, while other people don't. Even then, some people may like only particular short nicks to be highlighted (eg. names like "dan" are fine to highlight, whereas common words like "so", "the", "and" are not). What I'd suggest is to have a user-editable blacklist of words to not highlight, and maybe seed it with blacklist of common short English words ("the", "but", "and", "so", etc). I wasn't sure how opinionated the developers of this project would want to be, as having a default blacklist is quite opinionated.

**Comment by stevenengler** [comment]
I agree that the client should be smarter about this by default, but for those of us who want this feature, you can use the following CSS to blacklist common nicks:

```
.message .content .user[data-name="s"],
.message .content .user[data-name="t"] {
  color: inherit !important;
  pointer-events: none;
}
```

Just add it in the "Custom Stylesheet" section of the settings, and add more names as needed.

**Comment by ronilaukkarinen** [comment]
For me the CSS that actually works is:

```
#chat.colored-nicks .user[data-name="s"],
#chat.colored-nicks .user[data-name="t"] {
  color: inherit !important;
  pointer-events: none;
}
```

**Comment by itsjohncs** [comment]
I mentioned this on IRC yesterday, writing it here in case it captures someone's imagination enough to actually implement it. But we could use [`Intl.Segmenter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter) to improve the nick (and other) detection.

Firefox is the only browser that doesn't implement it yet but it seems like they will eventually ([bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1423593)).
